pkg/controller/deployment/sync.go

```diff
@@ -487,13 +533,22 @@ func (dc *DeploymentController) calculateStatus(allRSs []*extensions.ReplicaSet,
 	}
 	totalReplicas := deploymentutil.GetReplicaCountForReplicaSets(allRSs)
 
+	if availableReplicas >= deployment.Spec.Replicas-deploymentutil.MaxUnavailable(*deployment) {
```

Comment:
This is a big net win for clients to understand and be able to report on whether a deployment is usable under the definitions the deployment author set.